### PR TITLE
New code sample for dotnet/docs issue 12736

### PR DESCRIPTION
**Pull request by mssteele**
Part of a change to address issue:
https://github.com/dotnet/docs/issues/12736

My other change in the dotnet/docs repo contains the
corresponding documentation changes.
See PR: https://github.com/dotnet/docs/pull/17197

### CODE REVIEW

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 1]
```diff
@@ -1,49 +1,42 @@
...
     {
-       Title = title;    
-    }
+        int[] testSet = { 100271, 234335, 342439, 999683 };
 
-    public string Title { get; set; }
+        var primes = testSet.Where(n => Factor(n).ToList() is var factors
+                                    && factors.Count == 2
+                                    && factors.Contains(1)
+                                    && factors.Contains(n));
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]: The last two checks are not necessary: `1` and `n` are always factors of `n`. So, it looks those two checks are only for the sake of the example of the need of the `factors` variable?

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 2]
```diff
@@ -1,49 +1,42 @@
...
 
-    public override string ToString()
-    {
-       return Title;
+        foreach(int prime in primes)
+        {
+            Console.WriteLine("Found prime: " + prime);
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
            Console.WriteLine($"Found prime: {prime}");
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 3]
```diff
@@ -1,49 +1,42 @@
...
-   { get; set; }
-
-   public override string ToString()
-   {
-      return Name;
-   }
+    static IEnumerable<int> Factor(int number) 
+    {
+        int max = (int)Math.Sqrt(number);
+        for(int i = 1; i <= max; i++)
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
        for (int i = 1; i <= max; i++) 
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 4]
```diff
@@ -1,49 +1,42 @@
...
-   public override string ToString()
-   {
-      return Name;
-   }
+    static IEnumerable<int> Factor(int number) 
+    {
+        int max = (int)Math.Sqrt(number);
+        for(int i = 1; i <= max; i++) 
+        {
+            if(number % i == 0)
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
            if (number % i == 0)
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 5]
```diff
@@ -1,49 +1,42 @@
...
+            {
+                yield return i;
+                if(i != number/i)
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
                if (i != number / i) 
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 6]
```diff
@@ -1,49 +1,42 @@
...
-    public string Title { get; set; }
+        var primes = testSet.Where(n => Factor(n).ToList() is var factors
+                                    && factors.Count == 2
+                                    && factors.Contains(1)
+                                    && factors.Contains(n));
 
-    public override string ToString()
-    {
-       return Title;
+        foreach(int prime in primes)
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
        foreach (int prime in primes)
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs [hunk 7]
```diff
@@ -1,49 +1,42 @@
...
+    {
+        int max = (int)Math.Sqrt(number);
+        for(int i = 1; i <= max; i++) 
+        {
+            if(number % i == 0)
+            {
+                yield return i;
+                if(i != number/i) 
+                {
+                    yield return number/i;
```
Comment (pkulikov) [inline snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]:
```suggestion
                    yield return number / i;
```

File: snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs (hunk 1 above)
Comment (mssteele) [reply snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]: Good point! But as you say removing them means we don't need the `factors` variable which is the point of the example. 
I could change the example to do something else with the factors, say find numbers with between 3 and 5 factors, if you think this example is a bit misleading?

Comment (BillWagner) [reply snippets/csharp/language-reference/keywords/is/is-var-pattern8.cs]: I agree that the extra checks are unnecessary, but they do clarify the code.

I'm ready to approve this, pending the discussion on dotnet/docs#17197

**Pull request merged**